FlashDevelop/Managers/ScintillaManager.cs

```diff
@@ -20,14 +20,51 @@ namespace FlashDevelop.Managers
 {
     class ScintillaManager
     {
-        public static Scintilla SciConfig;
-        public static ConfigurationUtility SciConfigUtil;
+        public static event Action ConfigurationLoaded;
+
+        private static bool initialized;
+        private static readonly Object initializationLock = new Object();
+
         public static Bitmap Bookmark;
 
+        private static Scintilla _sciConfig;
+        public static Scintilla SciConfig
+        {
+            get
+            {
+                Initialize();
+                return _sciConfig;
+            }
+        }
+
+        private static ConfigurationUtility _sciConfigUtil;
+        public static ConfigurationUtility SciConfigUtil
+        {
+            get
+            {
+                Initialize();
+                return _sciConfigUtil;
+            }
+        }
+
         static ScintillaManager()
         {
             Bookmark = ScaleHelper.Scale(new Bitmap(ResourceHelper.GetStream("BookmarkIcon.png")));
-            LoadConfiguration();
+        }
+
+        private static void Initialize()
+        {
+            if (!initialized)
+            {
+                lock (initializationLock)
+                {
+                    if (!initialized)
+                    {
+                        LoadConfiguration();
```

Comment:
The previous static constructor was also doing the initialization in a deferred way, but with the problem that it was causing a big coupling between it and MainForm, which is not nice, and since it was a static constructor there was no way to set anything before, or calling any event handler or callback.

This ScintillaManager class looks like it could be in PluginCore after removing some other couplings with MainForm, I think some plugins could make use of it.